department-of-veterans-affairs/caseflow
# HunJerBAH/Appeals 4152

Resolves [APPEALS-4152](https://vajira.max.gov/browse/APPEALS-4152)

### Description
Added a POST function to the Split Appeal button in the Review Split page so that the appeal payload information could be sent to the backend for the split. 

### Acceptance Criteria
- [ ] When clicking "Split Appeal" on the Review and Confirm page of Split Appeal Workflow form data is submitted to backend url POST  /appeals/:appeal_id/split/

Form data included in the body includes the following:
- [ ] A list of issues that are moved to new appeal
- [ ] reason for split
- [ ] If Other was selected then include free text

- [ ] The call returns a status of success in order for the success banner to display on Case Details

### Testing Plan
1. Go to the **Switch Users** page on Caseflow and sign in as a Supervisory Senior Council user. 
![image](https://user-images.githubusercontent.com/99915461/185956501-bd266fde-fa64-4b6c-9b5d-2474ec73f444.png)
2. Navigate to **Queue** > **Search Cases**
![image](https://user-images.githubusercontent.com/99915461/185956739-550cb392-8ab6-4cb1-b095-bb5c2d42723e.png)
3. Put in a case number (500000000 works in most cases) and select the user below by clicking their docket number
![image](https://user-images.githubusercontent.com/99915461/185956891-0a9bbe39-45e7-469d-88d3-5f3f32c91fa9.png)
4. Click on **Correct Issues** link
![image](https://user-images.githubusercontent.com/99915461/185957145-6d7d7a4b-df33-4edf-a60a-04eb5e0d96fd.png)
5. If there are no issues on the case, add issues with the **Add issue** until there are at least 2 issues (the **Split appeal** button will show but be grayed out). Click **Save** to be redirected to the Queue page on **Step 4**. Once again, click the **Correct issues** link. The **Edit** page should now have the **Split appeal** button activated. 
![image](https://user-images.githubusercontent.com/99915461/185957871-65717459-2b04-40f7-9b06-dc5a573c7f47.png)
![image](https://user-images.githubusercontent.com/99915461/185958180-549423d0-2fdc-43d9-acf6-437e76a42046.png)

6. Click **Split appeal** to be directed to the **Create New Appeal Stream** page. Fill out the form and select at least 1 appeal to be split. The **Continue** button will then unlock. Click **Continue**.
![image](https://user-images.githubusercontent.com/99915461/185958369-a5f40652-255c-4dad-959f-05ef98d9c650.png)
![image](https://user-images.githubusercontent.com/99915461/185958443-2b36b9ce-70b3-4a79-a18a-4350f040e687.png)

7. On the **Review Split** page, open up the Inspector by right clicking your screen and selecting **Inspect** from the dropdown menu. Navigate to the **Rails** option for the inspector if you are using Chrome (you will need to download the Chrome Rails extension for this part if you do not have it). 
![image](https://user-images.githubusercontent.com/99915461/185962210-3671075e-307c-4290-98da-ed35e38a82e6.png)

8. Click the **Split Appeal** button. 
9. The page should load back to **Queue**. In the inspector, you should see that the **SplitAppealController#split_appeal** was contacted with a POST request. 
![image](https://user-images.githubusercontent.com/99915461/185962609-c5ea14dc-4912-43ea-b683-8c7cd87dc215.png)
